src/persistent_data.cpp

```diff
@@ -0,0 +1,50 @@
+#include "persistent_data.hpp"
```

Comment:
Licence is missing "// SPDX-License-Identifier: Apache-2.0" - The same applies to all newly added files.